relayer-engine/src/config/loadConfig.ts

```diff
@@ -42,9 +42,19 @@ export async function loadUntypedEnvs(
   };
 }
 
-async function loadCommon(dir: string, mode: Mode): Promise<any> {
+async function loadCommon(dir: string): Promise<any> {
   const obj = await loadFileAndParseToObject(`${dir}/common.json`);
-  obj.mode = mode;
+  if (obj.redis) {
+    if (process.env.RELAYER_ENGINE_REDIS_HOST) {
+      obj.redis.host = process.env.RELAYER_ENGINE_REDIS_HOST;
+    }
+    if (process.env.RELAYER_ENGINE_REDIS_USERNAME) {
+      obj.redis.username = process.env.RELAYER_ENGINE_REDIS_USERNAME;
+    }
+    if (process.env.RELAYER_ENGINE_REDIS_PASSWORD) {
+      obj.redis.password = process.env.RELAYER_ENGINE_REDIS_PASSWORD;
+    }
+  }
```

Comment:
I'm not proud of this.

Reply:
Generally not a fan of nested config objects, but can refine this later

Reply:
Agree, I figured we can revamp our config reading altogether since other things should be configurable via env vars too.

Reply:
I've not found a javascript as good as golang's [spf13/viper](https://github.com/spf13/viper), but [convict](https://github.com/mozilla/node-convict) is close. It allows command line flags, env vars, hardcoded defaults, and configuration files.